test/util/test_runners/test_runner_args.py

```diff
@@ -0,0 +1,44 @@
+from argparse import ArgumentParser
+
+
+class TestRunnerArgs:
+    """
+    Dto to hold the tests environment arguments
+    """
+
+    def __init__(self, hostname, account, username, password, invoke_cli_as_process, working_cli_path_only_for_init: str,
+                 cli_to_test_path: str, files_folder: str, test_name_identifier="integration"):
+        self.hostname = hostname
+        self.account = account
+        self.username = username
+        self.password = password
+        self.test_name_identifier = test_name_identifier
+        self.invoke_cli_as_process = invoke_cli_as_process
+        self.working_cli_path_only_for_init = working_cli_path_only_for_init
+        self.cli_to_test_path = cli_to_test_path
+        self.files_folder = files_folder
+        if cli_to_test_path is not None and  working_cli_path_only_for_init.lower() == cli_to_test_path.lower():
+            raise RuntimeError("cli to start enviroment and tested cli are on the same path")
+
+    @staticmethod
+    def create_from_args():
+        parser = ArgumentParser(description='Test arguments')
+
+        # Add the arguments
+        parser.add_argument('-invoke_cli_as_process', action='store_true',
+                            help='if using it, unitests will run cli process. otherwise it will run python code')
+        parser.add_argument('-i', '--identifier', dest='test_name_identifier', action='store', default='integration',
+                            help='if using it, unitests will run cli process. otherwise it will run python code')
+        parser.add_argument('--host', dest='hostname', action='store', default='https://conjur-https',
+                            help='server host name')
+        parser.add_argument('-a', '--account', dest='account', action='store', default='dev', help='account name')
+        parser.add_argument('-u', '--username', dest='username', action='store', default='admin', help='user name')
+        parser.add_argument('-p', '--password', dest='password', action='store', help='the user password')
+        parser.add_argument('-w', '--working_cli', dest='working_cli_path_only_for_init', action='store',
+                            help='the cli path using to set up the enviorment')
+        parser.add_argument('-c', '--cli_to_test_path', dest='cli_to_test_path', action='store',
+                            help='the cli binaries to test')
+        parser.add_argument('-f', '--files_folder', dest='files_folder', action='store', default='./test',
+                            help='where the conjurcc file is located')
+        args = parser.parse_args()
```

Comment:
We should be documenting in our contributing docs and to run the tests as a process. What do you think?

Reply:
absolutly right. show where and how?

Reply:
I would throw instructions under https://github.com/cyberark/conjur-api-python3/blob/master/CONTRIBUTING.md
In the codebase there is a file called[ CONTRIBUTING.md ](https://github.com/cyberark/conjur-api-python3/blob/master/CONTRIBUTING.md#testing) and I would put it under testing. Explaining we can run the CLI in one of two ways 1. as process 2. as code (requiring Python in the env). I would also give an example of how to run the different flows